### ISSUE TITLE
Conversion fails in SVG.py due to <image>

### ISSUE THREAD

**Issue by markthompson94** [hbmartin/graphviz2drawio#45]
```
Traceback (most recent call last):
  File "/usr/local/homebrew/bin/graphviz2drawio", line 8, in <module>
    sys.exit(main())
  File "/usr/local/homebrew/lib/python3.9/site-packages/graphviz2drawio/__main__.py", line 16, in main
    output = convert(args.to_convert, args.program)
  File "/usr/local/homebrew/lib/python3.9/site-packages/graphviz2drawio/graphviz2drawio.py", line 26, in convert
    nodes, edges = SvgParser(svg_graph).get_nodes_and_edges()
  File "/usr/local/homebrew/lib/python3.9/site-packages/graphviz2drawio/models/SvgParser.py", line 27, in get_nodes_and_edges
    nodes[title] = node_factory.from_svg(g)
  File "/usr/local/homebrew/lib/python3.9/site-packages/graphviz2drawio/mx/NodeFactory.py", line 60, in from_svg
    rect = self.rect_from_ellipse_svg(SVG.get_first(g, "ellipse").attrib)
  File "/usr/local/homebrew/lib/python3.9/site-packages/graphviz2drawio/models/SVG.py", line 2, in get_first
    return g.findall("./{http://www.w3.org/2000/svg}" + tag)[0]
IndexError: list index out of range
```

**Comment by markthompson94** [comment]
[digraph_example.docx](https://github.com/hbmartin/graphviz2drawio/files/8563464/digraph_example.docx)

Example .dot text included to assist troubleshooting

**Comment by markthompson94** [comment]
@hbmartin could you possibly take a look into this for us? Much love and thanks!!

**Comment by ams1** [comment]
just came across this really interesting repo -> @hbmartin THANK YOU for sharing!

@markthompson94 : i'll probably try to fix this over the weekend if it's still not working and get back to you.

**Comment by ams1** [comment]
@markthompson94 

The SVG file produced by graphviz via the Diagram lib (using the python code you provided) looks something like:

```xml
...

<!-- 1st g tag -->
<g id="edge3" class="edge">
  <title>94a1a24258c54e028e3b9cb0054dadc9&#45;&gt;739289249c574bffb2ed3d49b021b1da</title>
  <path fill="none" stroke="#7b8894" d="M338.5,-406.81C338.5,-406.81 338.5,-363.22 338.5,-363.22"/>

  <!-- this has a polygon tag -->
  <polygon fill="#7b8894" stroke="#7b8894" points="342,-363.22 338.5,-353.22 335,-363.22 342,-363.22"/>
</g>

...

<!-- 2nd g tag -->
<g id="node5" class="node">
  <title>2c1229f79a9141eea4b8ef297492ec26</title>
  <image xlink:href="does-not-matter/.local/lib/python3.8/site-packages/resources/aws/compute/ec2.png" width="101px" height="101px" preserveAspectRatio="xMinYMin meet" x="432" y="-335"/>
  <text text-anchor="middle" x="482.5" y="-219.6" font-family="Sans-Serif" font-size="13.00" fill="#2d3436">worker4</text>
  <!-- this does NOT hava a polygon tag -->
</g>

...
```

As we can see above, the 1st `<g ...>` tag HAS a `<polygon .../>` tag, while the 2nd g tag DOES NOT have a `<polygon .../>` tag, instead it has an `<image ... />` tag.

### Conclusion (might be wrong, I had just a quick look):
The problem is that this lib needs to have EITHER a `<polygon .../>` OR an `<ellipse.../>` in every `<g ...>` tag (see corresp. code below) and the python Diagrams returns an `<image ...>` tag tag instead, which is not supported.

https://github.com/hbmartin/graphviz2drawio/blob/751993505ef02db43ad116e7d4cd38a20608945a/graphviz2drawio/mx/NodeFactory.py#L55-L60

**Comment by markthompson94** [comment]
@hbmartin would you possibly be able to take a look at this? It'd be fantastic to have the ability to integrate Custom nodes from Python Diagrams in the graphviz2drawio module

**Comment by fabiomurillo** [comment]
Was anyone able to workaround this issue?

**Comment by markthompson94** [comment]
Think we need @hbmartin to help us out there 🤞

**Comment by markthompson94** [comment]
Hi @hbmartin, when you have any time would you be able to look into this at all?

thanks so much!!

**Comment by karth123** [comment]
@markthompson94 @fabiomurillo @ams1  I came across the repo for the same reason as you guys, wanted to render my diagrams file into drawio. I did resolve the svg issue after hacking around the library files a bit. (The solution is too long to share here). But even if you solve the svg render issue the edge placement is very poor and the actual XML generated is not satisfactory by any means.

I have created a script which does a better conversion into drawio SOLELY for diagrams library code. I know its 2 years too late but im putting it out there for anyone who had this issue like me.

https://github.com/karth123/diagrams-to-drawio

**Comment by hbmartin** [comment]
Work has started on this in https://github.com/hbmartin/graphviz2drawio/pull/49 and is slated for 0.4 release!

**Comment by hbmartin** [comment]
The fix for this will ship in 0.4 tomorrow, please reopen this issue if it doesn't meet your needs.

<img width="916" alt="Screenshot 2024-07-06 at 5 56 56 PM" src="https://github.com/hbmartin/graphviz2drawio/assets/745166/53e3244d-fa9b-40fa-b77a-dfeee31eee9a">
